### PR TITLE
Partially migrate EGEnergyCorrector to ESGetTokens and remove support for electrons

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

I was checking out how to migrate the `GEDPhotonProducer` to ESGetTokens.

One of the helper classes that must be migrated is the `EGEnergyCorrector` in EgammaTools. Besides some overall code modernization, this PR partially migrates the `EGEnergyCorrector` to ESGetTokens. It will be followed up with another PR to complete the migration, but I think it's better to do it step-by-step to keep the amount of changes to review at a reasonable size.

Since I noticed that the EGEnergyCorrector is not used for electrons anymore, the functions in this class that support electrons were removed. Electrons seem to exclusively use the newer regression modifiers.

#### PR validation:

CMSSW compiles and local matrix tests pass.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

No backport intended.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Review by jpata — COMMENTED** [review]
a very minor remark inline

**Comment by jpata** [comment]
DQM is picking up some very minor differences in tracks:

28234.0 -> SiOuterTrackerV/Tracks/FinalResolution/VtxZResolution
image

Given that tracks are not modified here, I suppose it's spurious?

File: RecoEgamma/EgammaPhotonProducers/src/GEDPhotonProducer.cc [hunk 1]
```diff
@@ -150,29 +146,27 @@ class GEDPhotonProducer : public edm::stream::EDProducer<> {
   double minR9Endcap_;
   bool runMIPTagger_;
 
-  bool validConversions_;
   RecoStepInfo recoStep_;
 
   bool usePrimaryVertex_;
   edm::ParameterSet conf_;
-  PositionCalc posCalculator_;
+  //PositionCalc posCalculator_;
```
[the remark this answers is not in the document]
Comment (guitargeek) [reply RecoEgamma/EgammaPhotonProducers/src/GEDPhotonProducer.cc:153]: I don't think so. I'll just delete these lines that I commented out so far, ok?

**Comment by guitargeek** [comment]
Good catch with the DQM diffs! They are absolutely not expected, and to me they even appear like some rounding issues.

**Comment by guitargeek** [comment]
Hi @jpata, thanks for the review. Only the most recent commit is actually new; I had to rebase on a new valid IB to test the code.

In the new commit, I address your comments and took the opportunity to do some more minor cleanup of the GEDPhotonProducer.

**Comment by qliphy** [comment]
please test

**Comment by jpata** [comment]
+reconstruction

- technical, modernizes EgammaPhotonProducers, EGEnergyCorrector and GEDPhotonProducer to use ESGetTokens
- no reco differences are expected or observed

**Comment by qliphy** [comment]
+1

**Pull request merged**